overlord/devicestate/devicemgr.go

```diff
@@ -1385,3 +1387,107 @@ func checkFDEFeatures(st *state.State, kernelInfo *snap.Info) error {
 	//       returns {"error":"hardware-unsupported"} we don't.
 	return nil
 }
+
+func (m *DeviceManager) runFDESetupHook(op string, params *boot.FdeSetupHookParams) ([]byte, error) {
+	// state is locked already by daemon
+	st := m.state
+
+	summary := fmt.Sprintf("Run fde-setup for %s", op)
+	chg := st.NewChange("fde-setup", summary)
+	hooksup := &hookstate.HookSetup{
+		Snap:     params.KernelInfo.InstanceName(),
+		Revision: params.KernelInfo.Revision,
+		Hook:     "fde-setup",
+		// XXX: should this be configurable somehow?
+		Timeout: 5 * time.Minute,
+	}
+	req := &hookstate.FDESetupRequest{
+		Op:      op,
+		Key:     params.Key,
+		KeyName: params.KeyName,
+		Model: map[string]string{
+			"series":     params.Model.Series(),
+			"brand-id":   params.Model.BrandID(),
+			"model":      params.Model.Model(),
+			"grade":      string(params.Model.Grade()),
+			"signkey-id": params.Model.SignKeyID(),
+		},
+		// TODO: include boot chains
+
+	}
+	contextData := map[string]interface{}{
+		"fde-setup-request": req,
+	}
+	task := hookstate.HookTask(st, summary, hooksup, contextData)
```

Comment:
This will have to use the new EphemeralRunHook() feature from #9715 - the approach here is wrong.

Reply:
This is updated now